File: backend/app/models/board.rb
class Board < ApplicationRecord
    belongs_to :owner, class_name: 'User', foreign_key: 'user_id'
    has_many :board_users, dependent: :destroy
    has_many :swim_lanes, dependent: :destroy
    has_many :collaborators, through: :board_users, source: :user
end

File: frontend/src/interfaces/Board.ts
import SwimLane from "./SwimLane";
import user from "./User";

export default interface Board {
    id: number;
    name: string;
    user_id: number;
    description: string;
    created_at: string;
    updated_at: string;
    // Included. Owner is child model of board.
    // (board belongs_to owner)
    owner: user;
    swim_lanes: SwimLane[];
}
